packages/design-system/src/components/N8nMenu/Menu.vue

```diff
@@ -26,55 +87,96 @@ export default Vue.extend({
 		defaultActive: {
 			type: String,
 		},
-		collapse: {
+		collapsed: {
 			type: Boolean,
+			default: false,
 		},
-		light: {
-			type: Boolean,
+		mode: {
+			type: String,
+			default: 'router',
+			validator: (value: string): boolean => ['router', 'tabs'].includes(value),
 		},
-		router: {
-			type: Boolean,
+		tooltipDelay: {
+			type: Number,
+			default: 300,
+		},
+		items: {
+			type: Array as PropType<IMenuItem[]>,
 		},
 	},
-	components: {
-		ElMenu, // eslint-disable-line @typescript-eslint/no-unsafe-assignment
+	mounted() {
+		if (this.mode === 'router') {
+			const found = this.items.find(item => {
+				return item.activateOnRouteNames !== undefined && (item.activateOnRouteNames ).includes(this.$route.name || '') ||
+					item.activateOnRoutePaths !== undefined && (item.activateOnRoutePaths).includes(this.$route.path);
+			});
+			this.activeTab = found ? found.id : '';
+		} else {
+			this.activeTab =  this.items.length > 0 ? this.items[0].id : '';
+		}
+	},
+	computed: {
+		upperMenuItems(): IMenuItem[] {
+			return this.items.filter((item: IMenuItem) => item.position === 'top' && item.available !== false);
```

Comment:
guessing about the name `item.available` is a `boolean` so
```suggestion
			return this.items.filter((item: IMenuItem) => item.position === 'top' && item.available);
```